### PR TITLE
Always enable nlohmann_json with examples

### PR DESCRIPTION
**Pull request by pratikvn**
Enable the file-config example conditionally, as it depends on nlohmann-json.

### CODE REVIEW

**Review by upsj — CHANGES_REQUESTED** [review]
find_package should ideally always happen on the topmost level, so that we don't look for different configurations accidentally. I would suggest moving the call to the root CMakeLists.txt. Is there a downside to always pulling in the dependency? I would like to avoid too many fragmented possible configurations depending on small environment differences.

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -64,6 +63,14 @@ else()
     message(STATUS "No Kokkos found, disabling examples with Kokkos assembly.")
 endif()
 
+set(GKO_FILE_CONFIG_ENABLED 0)
```
Comment (yhmtsai) [inline examples/CMakeLists.txt:66]: I think if you also add GINKGO_BUILD_EXAMPLES in the third_party CMakeLists, you do not need the change here

Comment (pratikvn) [reply examples/CMakeLists.txt:66]: Then we always pull in nlohmann_json when examples is built. But I guess that is okay.

**Review by yhmtsai — APPROVED** [review]

File: examples/CMakeLists.txt [hunk 2]
```diff
@@ -110,16 +110,15 @@ if(GINKGO_BUILD_TESTS)
...
-                     WORKING_DIRECTORY
-                     "$<TARGET_FILE_DIR:ginkgo>")
+            add_test(NAME example_file-config-solver_${config_name}_${executor}
+                COMMAND
+                "$<TARGET_FILE:file-config-solver>"
+                "${executor}" "${CMAKE_CURRENT_SOURCE_DIR}/file-config-solver/config/${config_name}.json"
+                "${CMAKE_CURRENT_SOURCE_DIR}/file-config-solver/data/A.mtx"
+                WORKING_DIRECTORY
+                "$<TARGET_FILE_DIR:ginkgo>")
             # Prevent performance issues with high core counts
```
Comment (upsj) [inline examples/CMakeLists.txt:120]: This doesn't actually change anything, right? I would just revert it then for a cleaner diff

**Review by upsj — APPROVED** [review]
LGTM!

**Pull request merged**